webapp/channels/src/components/markdown/index.ts

```diff
@@ -11,27 +13,79 @@ import {getPost} from 'mattermost-redux/selectors/entities/posts';
 import {getBool} from 'mattermost-redux/selectors/entities/preferences';
 import {getAllUserMentionKeys} from 'mattermost-redux/selectors/entities/search';
 import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+import {type UserHighlightKey} from 'mattermost-redux/selectors/entities/users';
 
 import {getEmojiMap} from 'selectors/emojis';
 
-import type {ChannelNamesMap, MentionKey} from 'utils/text_formatting';
+import type {TextFormattingOptions, MentionKey, ChannelNamesMap} from 'utils/text_formatting';
 import {getSiteURL} from 'utils/url';
 
 import type {GlobalState} from 'types/store';
 
 import Markdown from './markdown';
 
-type Props = {
+export type OwnProps = {
     channelNamesMap?: ChannelNamesMap;
     mentionKeys?: MentionKey[];
+    highlightKeys?: UserHighlightKey[];
     postId?: string;
+
+    /*
```

Comment:
I'm liking the `PropsFromRedux` pattern here, but I feel like it'd be better to leave the rest of these props in the component file instead of moving them here. That does mean that we might have to repeat a few fields when a component has an `OwnProps` field, but I think that's fine since most components don't use `OwnProps`. I think `connect` is actually a bit less performant if you use `ownProps` because it has to rerun `mapStateToProps` even if `state` doesn't change then.

Could you move the rest of these fields back to the `Props` type in the component file?

Reply:
done moved to the component